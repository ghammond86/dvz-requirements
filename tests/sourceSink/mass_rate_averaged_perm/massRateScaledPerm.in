#Description: 1D transient - fully saturated flow problem: checking mass rate - permeability and volume scaled

SIMULATION
  SIMULATION_TYPE SUBSURFACE
  PROCESS_MODELS
    SUBSURFACE_FLOW flow
      MODE RICHARDS
    /
  /
END

SUBSURFACE

#=========================== numerical methods ================================
NUMERICAL_METHODS FLOW

  LINEAR_SOLVER
    SOLVER DIRECT
  /
END

#=========================== solver options ===================================

#=========================== discretization ===================================
GRID
  TYPE UNSTRUCTURED_EXPLICIT ./irregular_1d_mesh.uge
END

#=========================== material properties ==============================
MATERIAL_PROPERTY soil1
  ID 1
  POROSITY 0.25
  ROCK_DENSITY 2500.0
  PERMEABILITY
    PERM_ISO 1e-14
  /
  CHARACTERISTIC_CURVES cc1
END

MATERIAL_PROPERTY soil2
  ID 2
  POROSITY 0.25
  ROCK_DENSITY 2500.0
  PERMEABILITY
    PERM_ISO 2e-14
  /
  CHARACTERISTIC_CURVES cc1
END

MATERIAL_PROPERTY soil3
  ID 3
  POROSITY 0.25
  ROCK_DENSITY 2500.0
  PERMEABILITY
    PERM_ISO 4e-14
  /
  CHARACTERISTIC_CURVES cc1
END

MATERIAL_PROPERTY soil4
  ID 4
  POROSITY 0.25
  ROCK_DENSITY 2500.0
  PERMEABILITY
    PERM_ISO 0.5e-14
  /
  CHARACTERISTIC_CURVES cc1
END

#=========================== characteristic curves ============================
CHARACTERISTIC_CURVES cc1
  SATURATION_FUNCTION VAN_GENUCHTEN
    ALPHA 1.d-4
    M 0.5d0
    LIQUID_RESIDUAL_SATURATION 0.1d0
  /
  PERMEABILITY_FUNCTION MUALEM_VG_LIQ
    M 0.5d0
    LIQUID_RESIDUAL_SATURATION 0.1d0
  /
END

#=========================== eos ==============================================
EOS WATER
 DENSITY CONSTANT 1000.0
 VISCOSITY CONSTANT 0.001017 Pa-s
END

#=========================== output options ===================================
OUTPUT
  MASS_BALANCE_FILE
  	PERIODIC TIMESTEP 1
  /
  
END

#=========================== times ============================================
TIME
  FINAL_TIME 1.0 d
  INITIAL_TIMESTEP_SIZE 0.1 d
  MAXIMUM_TIMESTEP_SIZE 0.1 d
END

#=========================== regions ==========================================
REGION all
  COORDINATES
    0.0 0.0 0.0
    2.0 1.0 7.5
  /
END

REGION injector
  COORDINATES
    0.0 0.0 0.0
    1.0 1.0 7.5
  /
END

REGION cell1
	FILE cell_1_bc.ex
END

REGION cell2
	FILE cell_2_bc.ex
END

REGION cell3
	FILE cell_3_bc.ex
END

REGION cell4
	FILE cell_4_bc.ex
END

REGION soil1
    COORDINATES
      0.0 0.0 0.0
      2.0 1.0 1.0
    /
END

REGION soil2
    COORDINATES
      0.0 0.0 1.0
      2.0 1.0 3.0
    /
END

REGION soil3
    COORDINATES
      0.0 0.0 3.0
      2.0 1.0 7.0
    /
END

REGION soil4
    COORDINATES
      0.0 0.0 7.0
      2.0 1.0 7.5
    /
END

#=========================== flow conditions ==================================
FLOW_CONDITION initial_pressure
  TYPE
    PRESSURE DIRICHLET
  /
  PRESSURE 101325.0 Pa
END

FLOW_CONDITION right_face
  TYPE
    PRESSURE DIRICHLET
  /
  PRESSURE 101325.0 Pa
END

FLOW_CONDITION injector
  TYPE
    RATE SCALED_MASS_RATE PERM
  / 
  RATE 1.d-3 g/s
END

#=========================== condition couplers ===============================
INITIAL_CONDITION
  FLOW_CONDITION initial_pressure
  REGION all
END

BOUNDARY_CONDITION cell1
  FLOW_CONDITION right_face
  REGION cell1
END

BOUNDARY_CONDITION cell2
  FLOW_CONDITION right_face
  REGION cell2
END

BOUNDARY_CONDITION cell3
  FLOW_CONDITION right_face
  REGION cell3
END

BOUNDARY_CONDITION cell4
  FLOW_CONDITION right_face
  REGION cell4
END

SOURCE_SINK injector
  FLOW_CONDITION injector
  REGION injector
END

#=========================== stratigraphy couplers ============================
STRATA
  REGION soil1
  MATERIAL soil1
END

STRATA
  REGION soil2
  MATERIAL soil2
END

STRATA
  REGION soil3
  MATERIAL soil3
END

STRATA
  REGION soil4
  MATERIAL soil4
END

END_SUBSURFACE
