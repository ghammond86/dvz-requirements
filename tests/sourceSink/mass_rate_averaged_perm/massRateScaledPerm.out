
=================
  PFLOTRAN v3.0
=================

 Extent of Gridded Domain
  X: 0.50000000000000000 - 1.5000000000000000
  Y: 0.50000000000000000 - 0.50000000000000000
  Z: 0.50000000000000000 - 7.2500000000000000


=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+

  Initial Condition

--------------------------------------------------------------------------------
          Flow Condition:   initial_pressure
                  Region:   all
--------------------------------------------------------------------------------

  Flow Condition: initial_pressure
    Synchronize time with update: no
    Time units: yr
    Length units: m
      Datum:

    Sub Condition: pressure
      Type: dirichlet
      Gradient:
      Data:
        Dataset: Unnamed Ascii Dataset
          Type: dataset_ascii_type
          Transient: no
          Array Rank:  1

--------------------------------------------------------------------------------

=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+

  Boundary Condition

--------------------------------------------------------------------------------
          Flow Condition:   right_face
                  Region:   cell1
--------------------------------------------------------------------------------

  Flow Condition: right_face
    Synchronize time with update: no
    Time units: yr
    Length units: m
      Datum:

    Sub Condition: pressure
      Type: dirichlet
      Gradient:
      Data:
        Dataset: Unnamed Ascii Dataset
          Type: dataset_ascii_type
          Transient: no
          Array Rank:  1

--------------------------------------------------------------------------------

=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+

  Boundary Condition

--------------------------------------------------------------------------------
          Flow Condition:   right_face
                  Region:   cell2
--------------------------------------------------------------------------------

  Flow Condition: right_face
    Synchronize time with update: no
    Time units: yr
    Length units: m
      Datum:

    Sub Condition: pressure
      Type: dirichlet
      Gradient:
      Data:
        Dataset: Unnamed Ascii Dataset
          Type: dataset_ascii_type
          Transient: no
          Array Rank:  1

--------------------------------------------------------------------------------

=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+

  Boundary Condition

--------------------------------------------------------------------------------
          Flow Condition:   right_face
                  Region:   cell3
--------------------------------------------------------------------------------

  Flow Condition: right_face
    Synchronize time with update: no
    Time units: yr
    Length units: m
      Datum:

    Sub Condition: pressure
      Type: dirichlet
      Gradient:
      Data:
        Dataset: Unnamed Ascii Dataset
          Type: dataset_ascii_type
          Transient: no
          Array Rank:  1

--------------------------------------------------------------------------------

=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+

  Boundary Condition

--------------------------------------------------------------------------------
          Flow Condition:   right_face
                  Region:   cell4
--------------------------------------------------------------------------------

  Flow Condition: right_face
    Synchronize time with update: no
    Time units: yr
    Length units: m
      Datum:

    Sub Condition: pressure
      Type: dirichlet
      Gradient:
      Data:
        Dataset: Unnamed Ascii Dataset
          Type: dataset_ascii_type
          Transient: no
          Array Rank:  1

--------------------------------------------------------------------------------

=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+

  Source Sink

--------------------------------------------------------------------------------
          Flow Condition:   injector
                  Region:   injector
--------------------------------------------------------------------------------

  Flow Condition: injector
    Synchronize time with update: no
    Time units: yr
    Length units: m
      Datum:

    Sub Condition: rate
      Type: scaled mass rate
      Gradient:
      Data:
        Dataset: Unnamed Ascii Dataset
          Type: dataset_ascii_type
          Transient: no
          Array Rank:  1

--------------------------------------------------------------------------------

 Requested processors =     1

FLOW Time Stepper
                 acceleration: on
       acceleration threshold: 5
       number of ramp entries: 5
                ramp entry #1: 2.0000
                ramp entry #2: 2.0000
                ramp entry #3: 2.0000
                ramp entry #4: 2.0000
                ramp entry #5: 2.0000
                ramp entry #6: 1.8000
                ramp entry #7: 1.6000
                ramp entry #8: 1.4000
                ramp entry #9: 1.2000
               ramp entry #10: 1.0000
               ramp entry #11: 1.0000
               ramp entry #12: 1.0000
               ramp entry #13: 1.0000
      maximum number of steps: 999999
constant time steps threshold: 5
       maximum number of cuts: 16
             reduction factor: 5.000000E-01
        maximum growth factor: 2.000000E+00

FLOW Newton Solver
                         atol: 1.000000E-50
                         rtol: 1.000000E-08
                         stol: 1.000000E-08
                         dtol: 1.000000E+04
                      maxnorm: 1.000000E+20
                    inftolres: -9.990000E+02
                    inftolupd: -9.990000E+02
                 inftolrelupd: -9.990000E+02
                 inftolsclres: -9.990000E+02
                     max iter: 50
                     min iter: 1
                         maxf: 10000
                  matrix type: baij
         precond. matrix type: baij
               inexact newton: off
            print convergence: on
   print detailed convergence: off
          check infinity norm: on

FLOW Linear Solver
                       solver: preonly
               preconditioner: lu
                         atol: 1.000000E-50
                         rtol: 1.000000E-05
                         dtol: 1.000000E+04
            maximum iteration: 10000

== RICHARDS FLOW ===============================================================
  0 2r: 4.32E-08 2x: 0.00E+00 2u: 0.00E+00 ir: 4.18E-08 iu: 0.00E+00 rsn:   0
  1 2r: 2.91E-19 2x: 2.87E+05 2u: 2.03E+01 ir: 2.34E-19 iu: 7.18E+00 rsn: rtol
 Step      1 Time=  1.00000E-01 Dt=  1.00000E-01 [d] snes_conv_reason:    3
  newton =   1 [       1] linear =     1 [         1] cuts =  0 [   0]
  --> max chng: dpmx=   7.1788E+00

== RICHARDS FLOW ===============================================================
  0 2r: 2.91E-19 2x: 0.00E+00 2u: 0.00E+00 ir: 2.34E-19 iu: 0.00E+00 rsn:   0
  1 2r: 7.06E-19 2x: 2.87E+05 2u: 4.12E-10 ir: 6.33E-19 iu: 1.89E-10 rsn: stol
 Step      2 Time=  2.00000E-01 Dt=  1.00000E-01 [d] snes_conv_reason:    4
  newton =   1 [       2] linear =     1 [         2] cuts =  0 [   0]
  --> max chng: dpmx=   1.8917E-10

== RICHARDS FLOW ===============================================================
  0 2r: 7.06E-19 2x: 0.00E+00 2u: 0.00E+00 ir: 6.33E-19 iu: 0.00E+00 rsn:   0
  1 2r: 2.91E-19 2x: 2.87E+05 2u: 7.05E-10 ir: 2.34E-19 iu: 4.07E-10 rsn: stol
 Step      3 Time=  3.00000E-01 Dt=  1.00000E-01 [d] snes_conv_reason:    4
  newton =   1 [       3] linear =     1 [         3] cuts =  0 [   0]
  --> max chng: dpmx=   4.0745E-10

== RICHARDS FLOW ===============================================================
  0 2r: 2.91E-19 2x: 0.00E+00 2u: 0.00E+00 ir: 2.34E-19 iu: 0.00E+00 rsn:   0
  1 2r: 3.05E-19 2x: 2.87E+05 2u: 4.12E-10 ir: 2.34E-19 iu: 1.89E-10 rsn: stol
 Step      4 Time=  4.00000E-01 Dt=  1.00000E-01 [d] snes_conv_reason:    4
  newton =   1 [       4] linear =     1 [         4] cuts =  0 [   0]
  --> max chng: dpmx=   1.8917E-10

== RICHARDS FLOW ===============================================================
  0 2r: 3.05E-19 2x: 0.00E+00 2u: 0.00E+00 ir: 2.34E-19 iu: 0.00E+00 rsn:   0
  1 2r: 3.91E-19 2x: 2.87E+05 2u: 1.45E-09 ir: 2.75E-19 iu: 1.00E-09 rsn: stol
 Step      5 Time=  5.00000E-01 Dt=  1.00000E-01 [d] snes_conv_reason:    4
  newton =   1 [       5] linear =     1 [         5] cuts =  0 [   0]
  --> max chng: dpmx=   1.0041E-09

== RICHARDS FLOW ===============================================================
  0 2r: 3.91E-19 2x: 0.00E+00 2u: 0.00E+00 ir: 2.75E-19 iu: 0.00E+00 rsn:   0
  1 2r: 6.57E-19 2x: 2.87E+05 2u: 6.91E-10 ir: 6.33E-19 iu: 4.07E-10 rsn: stol
 Step      6 Time=  6.00000E-01 Dt=  1.00000E-01 [d] snes_conv_reason:    4
  newton =   1 [       6] linear =     1 [         6] cuts =  0 [   0]
  --> max chng: dpmx=   4.0745E-10

== RICHARDS FLOW ===============================================================
  0 2r: 6.57E-19 2x: 0.00E+00 2u: 0.00E+00 ir: 6.33E-19 iu: 0.00E+00 rsn:   0
  1 2r: 2.91E-19 2x: 2.87E+05 2u: 4.36E-10 ir: 2.34E-19 iu: 1.89E-10 rsn: stol
 Step      7 Time=  7.00000E-01 Dt=  1.00000E-01 [d] snes_conv_reason:    4
  newton =   1 [       7] linear =     1 [         7] cuts =  0 [   0]
  --> max chng: dpmx=   1.8917E-10

== RICHARDS FLOW ===============================================================
  0 2r: 2.91E-19 2x: 0.00E+00 2u: 0.00E+00 ir: 2.34E-19 iu: 0.00E+00 rsn:   0
  1 2r: 3.91E-19 2x: 2.87E+05 2u: 4.12E-10 ir: 2.75E-19 iu: 1.89E-10 rsn: stol
 Step      8 Time=  8.00000E-01 Dt=  1.00000E-01 [d] snes_conv_reason:    4
  newton =   1 [       8] linear =     1 [         8] cuts =  0 [   0]
  --> max chng: dpmx=   1.8917E-10

== RICHARDS FLOW ===============================================================
  0 2r: 3.91E-19 2x: 0.00E+00 2u: 0.00E+00 ir: 2.75E-19 iu: 0.00E+00 rsn:   0
  1 2r: 6.57E-19 2x: 2.87E+05 2u: 6.91E-10 ir: 6.33E-19 iu: 4.07E-10 rsn: stol
 Step      9 Time=  9.00000E-01 Dt=  1.00000E-01 [d] snes_conv_reason:    4
  newton =   1 [       9] linear =     1 [         9] cuts =  0 [   0]
  --> max chng: dpmx=   4.0745E-10

== RICHARDS FLOW ===============================================================
  0 2r: 6.57E-19 2x: 0.00E+00 2u: 0.00E+00 ir: 6.33E-19 iu: 0.00E+00 rsn:   0
  1 2r: 3.79E-19 2x: 2.87E+05 2u: 4.36E-10 ir: 2.75E-19 iu: 1.89E-10 rsn: stol
 Step     10 Time=  1.00000E+00 Dt=  1.00000E-01 [d] snes_conv_reason:    4
  newton =   1 [      10] linear =     1 [        10] cuts =  0 [   0]
  --> max chng: dpmx=   1.8917E-10

 Wall Clock Time:  7.3227E-02 [sec]   1.2204E-03 [min]   2.0341E-05 [hr]
